packages/react-ui/src/app/builder/piece-properties/multi-select-piece-property.tsx

```diff
@@ -23,79 +23,114 @@ type MultiSelectPiecePropertyProps = {
   onChange: (value: unknown[]) => void;
   initialValues?: unknown[];
   disabled?: boolean;
+  enableSelectOrClear?: boolean;
 };
 
+const SELECT_OR_CLEAR_MIN_OPTIONS = 3;
+
 const MultiSelectPieceProperty = ({
   placeholder,
   options,
   onChange,
   disabled,
   initialValues,
+  enableSelectOrClear,
 }: MultiSelectPiecePropertyProps) => {
-  const [originalOptionsWithIndexes] = useState(
-    options.map((option, index) => ({
-      ...option,
-      originalIndex: index,
-    })),
-  );
-  const [searchTerm, setSearchTerm] = useState('');
-  const [filteredOptions, setFilteredOptions] = useState(
-    originalOptionsWithIndexes,
-  );
-  useEffect(() => {
-    setFilteredOptions(
-      originalOptionsWithIndexes.filter((option) => {
-        return option.label.toLowerCase().includes(searchTerm.toLowerCase());
-      }),
-    );
-  }, [searchTerm, originalOptionsWithIndexes]);
-  const selectedIndicies = initialValues
-    ? initialValues
-        .map((value) =>
-          options.findIndex((option) => deepEqual(option.value, value)),
-        )
-        .filter((index) => index > -1)
-        .map((index) => String(index))
-    : [];
-  const sendChanges = (indicides: string[]) => {
-    const newSelectedIndicies = indicides.filter(
-      (index) => index !== undefined,
-    );
-    onChange(newSelectedIndicies.map((index) => options[Number(index)].value));
+  const selectClearEnabled =
+    enableSelectOrClear && options.length > SELECT_OR_CLEAR_MIN_OPTIONS;
+
+  const onSelectOrClearChanged = (changeType: SelectOrClearChangeType) => {
+    if (changeType === 'selectAll') {
+      onChange(options.map((o) => String(o.value)));
+    } else {
+      onChange([]);
+    }
   };
 
   return (
     <MultiSelect
-      modal={true}
-      value={selectedIndicies}
-      onValueChange={sendChanges}
+      value={initialValues as string[]}
+      onValueChange={onChange}
       disabled={disabled}
-      onSearch={(searchTerm) => setSearchTerm(searchTerm ?? '')}
+      filter={true}
```

Comment:
I know the implementation has diverged from the original, this is because it's easier to deal with values than indexes, and filtering is already enabled by the MultiSelect component.